orderbook/src/fee.rs

```diff
@@ -0,0 +1,178 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use chrono::{DateTime, Duration, Utc};
+use primitive_types::{H160, U256};
+use std::sync::Mutex;
+
+use crate::price_estimate::PriceEstimating;
+use gas_estimation::GasPriceEstimating;
+
+type Measurement = (U256, DateTime<Utc>);
+
+pub struct MinFeeCalculator {
+    price_estimator: Box<dyn PriceEstimating>,
+    gas_estimator: Box<dyn GasPriceEstimating>,
+    native_token: H160,
+    // TODO persist past measurements to shared storage
+    measurements: Mutex<HashMap<H160, Vec<Measurement>>>,
+    now: Box<dyn Fn() -> DateTime<Utc>>,
+}
+
+const GAS_PER_ORDER: f64 = 100_000.0;
+const STANDARD_VALIDITY_FOR_FEE_IN_SEC: i64 = 60;
```

Comment:
Now you can make this a real `Duration` too.

Reply:
Duration::seconds doesn't seem to be `const`. Probably not worth using a lazy_static here?

Reply:
Agree not worth using lazy static but I thought it was const https://doc.rust-lang.org/std/time/struct.Duration.html#method.from_secs

Reply:
We are using `chrono::DateTime` and [`chrono::Duration::seconds`](https://docs.rs/chrono/0.4.19/chrono/struct.Duration.html#method.seconds) is not static

Reply:
Could create the chrono Duration from the std Duration when used. But anyway not important at all and not blocking the PR.